src/abstract/CSBondCurve.sol

```diff
@@ -49,12 +49,17 @@ abstract contract CSBondCurve is ICSBondCurve, Initializable {
         MAX_CURVE_LENGTH = maxCurveLength;
     }
 
+    /// @inheritdoc ICSBondCurve
+    function curveExists(uint256 curveId) public view returns (bool) {
+        return curveId < _getCSBondCurveStorage().bondCurves.length;
+    }
+
     /// @inheritdoc ICSBondCurve
     function getCurveInfo(
         uint256 curveId
     ) public view returns (BondCurve memory) {
         CSBondCurveStorage storage $ = _getCSBondCurveStorage();
-        if (curveId > $.bondCurves.length - 1) revert InvalidBondCurveId();
```

Comment:
I think would be better to have `_curveExists(CSBondCurveStorage _storage)` to not to read the storage twice

Reply:
I think `_getCSBondCurveStorage()` doesn't read from the storage, and just sets the slot value for the future access instead.

Reply:
I think doesn't really mater, for me it's ok as well with private func as it is now